pages/_app.js

```diff
@@ -103,6 +111,16 @@ export default function App({ Component, pageProps }) {
     theme === "light" ? setTheme("dark") : setTheme("light");
   }
 
+  function handleNextPage() {
+    setResultsPage(resultsPage + 1);
+  }
+
+  function handlePrevPage() {
+    if (resultsPage > 1) {
+      setResultsPage(resultsPage - 1);
+    }
+  }
```

Comment:
You should prevent to have a negative pageIndex here. Additionally, if a state change uses a previous state value, it's always safer to use the callback form of the state setter: `setResultsPage((previous)=>previous-1)`;